runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -679,32 +714,31 @@ protected void syncStreamUnsafe(ISMRStream stream, long timestamp) {
 
-        Histogram histogram = VloMetricsHelper.metrics.histogram(syncStreamCount);
-        Counter counter = new Counter();
-        try (Timer.Context context = getVloStreamSyncContext(timestamp, syncStreamTimer)) {
-            stream.streamUpTo(syncTo)
-                    .forEachOrdered(entry -> {
-                        try {
-                            counter.count++;
-                            Object res = applyUpdateUnsafe(entry, timestamp);
-                            if (timestamp == Address.OPTIMISTIC) {
+        Runnable syncStreamRunnable = () ->
+                stream.streamUpTo(syncTo)
+                        .forEachOrdered(entry -> {
+                            try {
+                                Object res = applyUpdateUnsafe(entry, timestamp);
+                                versionApplyCounter.ifPresent(Counter::increment);
+                                if (timestamp == Address.OPTIMISTIC) {
+                                    entry.setUpcallResult(res);
+                                } else if (pendingUpcalls.contains(entry.getGlobalAddress())) {
+                                    log.debug("Sync[{}] Upcall Result {}",
+                                            this, entry.getGlobalAddress());
+                                    upcallResults.put(entry.getGlobalAddress(), res == null
+                                            ? NullValue.NULL_VALUE : res);
+                                    pendingUpcalls.remove(entry.getGlobalAddress());
+                                }
                                 entry.setUpcallResult(res);
-                            } else if (pendingUpcalls.contains(entry.getGlobalAddress())) {
-                                log.debug("Sync[{}] Upcall Result {}",
-                                        this, entry.getGlobalAddress());
-                                upcallResults.put(entry.getGlobalAddress(), res == null
-                                        ? NullValue.NULL_VALUE : res);
-                                pendingUpcalls.remove(entry.getGlobalAddress());
+                            } catch (Exception e) {
+                                log.error("Sync[{}] Error: Couldn't execute upcall due to {}", this, e);
+                                throw new UnrecoverableCorfuError(e);
                             }
-                            entry.setUpcallResult(res);
-                        } catch (Exception e) {
-                            log.error("Sync[{}] Error: Couldn't execute upcall due to {}", this, e);
-                            throw new UnrecoverableCorfuError(e);
-                        }
-                    });
-        }
-        if (timestamp != Address.OPTIMISTIC) {
-            histogram.update(counter.count);
+                        });
+        if (syncStreamTimer.isPresent()) {
+            syncStreamTimer.get().record(syncStreamRunnable);
```

Comment:
Not sure if we need to use 2 different timers to record optimistic sync and regular sync